nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{correct_name}'"
```

Comment:
- [ ] Please be aware that this line will run infinite times because the method is calling itself recursively at the end of the error message! 😱 To solve this, please consider using `'#{__method__}'` instead of `'#{correct_name}'` as `__method__` returns the name of the current method. 🤓 🔧